File: src/test/java/com/antero/liemipeli/domain/opiskelu/KurssitoteutusTest.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.antero.liemipeli.domain.opiskelu;

import java.util.ArrayList;
import java.util.Collection;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;

/**
 *
 * @author Antero Oikkonen
 */
public class KurssitoteutusTest {

    Kurssitoteutus toteutus = null;
    
    public KurssitoteutusTest() {
    }

    @Before
    public void setUp() {
        toteutus=new Kurssitoteutus();
        toteutus.setKurssi(luoKurssi("Liementeon perusteet I"));
        toteutus.setKurssinKokonaisPisteet(100);
        toteutus.setKurssinEteneminen(0);
    }

    @Test
    public void testGetHarjoitukset() {
        System.out.println("getHarjoitukset");
        Kurssitoteutus instance = new Kurssitoteutus();
        Collection<Harjoitus> expResult = new ArrayList();
        Harjoitus h = new Harjoitus("Mitä on 15/3", "5");
        expResult.add(h);
        instance.setHarjoitukset(expResult);
        Collection<Harjoitus> result = instance.getHarjoitukset();

        assertEquals(expResult, result);
        assertEquals(1, result.size());
        assertEquals("5", ((Harjoitus) (result.iterator().next())).getVastaus());

        //fail("The test case is a prototype.");
    }

    @Test
    public void testPuuttuvatPisteet() {
        System.out.println("PuuttuvatPisteet");
      
        toteutus.opiskele();
        toteutus.opiskele();
        assertEquals(80, toteutus.montaPistettaPuuttuu());
    }

    @Test
    public void testOnkoKurssiSuoritettuJosEiOle() {
        System.out.println("OnkoKurssiSuoritettuJosEiOle");
        Kurssi k = luoKurssi("Taiat I");

        Kurssitoteutus instance = new Kurssitoteutus();
        instance.setKurssi(k);
        instance.setKurssinKokonaisPisteet(100);
        instance.opiskele();

        assertEquals(false, instance.isSuoritettu());
    }

    @Test
    public void testOnkoKurssiSuoritettuJosOn() {
        System.out.println("OnkoKurssiSuoritettuJosEiOle");
        Kurssi k = luoKurssi("Taiat I");

        Kurssitoteutus instance = new Kurssitoteutus();
        instance.setKurssi(k);
        instance.setKurssinKokonaisPisteet(100);
        for (int i = 0; i < 10; i++) {
            instance.opiskele();
        }

        assertEquals(true, instance.isSuoritettu());
    }

    @Test
    public void testOpiskele() {
        System.out.println("opiskele");
        Kurssi k = luoKurssi("Taiat I");

        Kurssitoteutus instance = new Kurssitoteutus();
        instance.setKurssi(k);
        instance.setKurssinKokonaisPisteet(100);
        instance.opiskele();

        assertEquals(10, instance.getKurssinEteneminen());

    }

    @Test
    public void testGetOpettaja() {
        System.out.println("getOpettaja");
        Kurssitoteutus instance = new Kurssitoteutus();
        Opettaja expResult = new Opettaja();
        expResult.setOppiarvo("Liementeon mestari");
        instance.setOpettaja(expResult);
        Opettaja result = instance.getOpettaja();
        assertEquals(expResult, result);
        //fail("The test case is a prototype.");
    }

    @Test
    public void testSetKurssi() {
        System.out.println("setKurssi");
        String kurssinNimi = "Liementeko I";
        Kurssi kurssi = luoKurssi(kurssinNimi);

        Kurssitoteutus instance = new Kurssitoteutus();
        instance.setKurssi(kurssi);
        Kurssi k = instance.getKurssi();
        assertEquals(10, k.getOpintopisteet());
        assertEquals(1000, k.getKurssinHinta());
        assertEquals(kurssinNimi, k.getKurssinNimi());
        //fail("The test case is a prototype.");
    }

    private Kurssi luoKurssi(String kurssinNimi) {

        Kurssi kurssi = new Kurssi();
        kurssi.setKurssinHinta(1000);
        kurssi.setKurssinNimi(kurssinNimi);
        kurssi.setOpintopisteet(10);
        return kurssi;
    }

}

File: liemipeli-kotlin/src/test/kotlin/com/antero/liemipeli/domain/opiskelu/KurssitoteutusTest.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.antero.liemipeli.domain.opiskelu

import com.antero.liemipeli.opiskelu.Harjoitus
import com.antero.liemipeli.opiskelu.Kurssi
import com.antero.liemipeli.opiskelu.Kurssitoteutus
import com.antero.liemipeli.opiskelu.Opettaja
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.fail
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance
import org.junit.jupiter.api.TestInstance.Lifecycle


/**
 *
 * @author Antero Oikkonen
 */
@TestInstance(Lifecycle.PER_CLASS)
class KurssitoteutusTest {

    internal var toteutus: Kurssitoteutus? = null

    @BeforeAll
    fun setUp() {
        toteutus = Kurssitoteutus()
        toteutus?.kurssi = luoKurssi("Liementeon perusteet I")
        toteutus?.kurssinKokonaisPisteet = 100
        toteutus?.kurssinEteneminen = 0
    }

    @Test
    fun testGetHarjoitukset() {
        println("getHarjoitukset")
        val instance = Kurssitoteutus()
        val expResult = ArrayList<Harjoitus>()
        val h = Harjoitus("Mitä on 15/3", "5")
        expResult.add(h)
        instance.harjoitukset = expResult
        val result = instance.harjoitukset

        assertEquals(expResult, result)
        assertEquals(1, result?.size)
        assertEquals("5", (result?.iterator()?.next() as Harjoitus).vastaus)
    }

    @Test
    fun testPuuttuvatPisteet() {
        println("PuuttuvatPisteet")

        toteutus!!.opiskele()
        toteutus!!.opiskele()
        assertEquals(80, toteutus!!.montaPistettaPuuttuu())
    }

    @Test
    fun testOnkoKurssiSuoritettuJosEiOle() {
        println("OnkoKurssiSuoritettuJosEiOle")
        val k = luoKurssi("Taiat I")

        val instance = Kurssitoteutus()
        instance.kurssi = k
        instance.kurssinKokonaisPisteet = 100
        instance.opiskele()

        assertEquals(false, instance.isSuoritettu)
    }

    @Test
    fun testOnkoKurssiSuoritettuJosOn() {
        println("OnkoKurssiSuoritettuJosEiOle")
        val k = luoKurssi("Taiat I")

        val instance = Kurssitoteutus()
        instance.kurssi = k
        instance.kurssinKokonaisPisteet = 100
        for (i in 0..9) {
            instance.opiskele()
        }

        assertEquals(true, instance.isSuoritettu)
    }

    @Test
    fun testOpiskele() {
        println("opiskele")
        val k = luoKurssi("Taiat I")

        val instance = Kurssitoteutus()
        instance.kurssi = k
        instance.kurssinKokonaisPisteet = 100
        instance.opiskele()

        assertEquals(10, instance.kurssinEteneminen)

    }

    @Test
    fun testGetOpettaja() {
        println("getOpettaja")
        val instance = Kurssitoteutus()
        val expResult = Opettaja()
        expResult.oppiarvo = "Liementeon mestari"
        instance.opettaja = expResult
        val result = instance.opettaja
        assertEquals(expResult, result)
        //fail("The test case is a prototype.");
    }

    @Test
    fun testSetKurssi() {
        println("setKurssi")
        val kurssinNimi = "Liementeko I"
        val kurssi = luoKurssi(kurssinNimi)

        val instance = Kurssitoteutus()
        instance.kurssi = kurssi
        val k = instance.kurssi
        assertEquals(10, k?.opintopisteet)
        assertEquals(1000, k?.kurssinHinta)
        assertEquals(kurssinNimi, k?.kurssinNimi)
        //fail("The test case is a prototype.");
    }

    private fun luoKurssi(kurssinNimi: String): Kurssi {

        val kurssi = Kurssi()
        kurssi.kurssinHinta = 1000
        kurssi.kurssinNimi = kurssinNimi
        kurssi.opintopisteet = 10
        return kurssi
    }

}
